craft_application/services/lifecycle.py

```diff
@@ -225,14 +225,22 @@ def _init_lifecycle_manager(self) -> LifecycleManager:
         emit.debug(f"Project vars: {self._project_vars}")
         emit.debug(f"Adopting part: {self._project.adopt_info}")
 
+        source_ignore_patterns = [
+            *self._app.source_ignore_patterns,
+            # Ignore spread.yaml and spread to prevent repulling sources
+            # when test files are changed.
+            "spread.yaml",
+            "spread",
```

Comment:
I'm fine with `spread.yaml` but a bit more concerned about `spread`. There seem to be a lot of projects on GitHub that use the term `spread` for a directory name.

Reply:
Also, are either of these relocatable?  I recall a discussion about enhancing spread to be able to point to a `spread.yaml` with an alternate name or location.

Reply:
Spread has that, but it specifically exists to support craft test. (We create a tempdir and put our processed spread.yaml inside it.) 

That does raise a question though: could we have this apply only to `spread.yaml` in the top directory?

Reply:
We can apply only to `spread.yaml`, however it defeats the purpose of not rebuilding if the tests are changed. Maybe we could check the presence of `spread/.extension`, and ignore the directory in this case?